### DISCUSSION TITLE
Index bar chart by key, but with different label

### DISCUSSION THREAD

**Discussion by melarenina** [plouc/nivo#1716] [Q&A] [answered]
Hey,

Currently, I have this bar chart, with the `indexBy` using a value called `carrierId` (which are those numbers marked as red). But, doing this, my chart does not get sorted by value (first is the one with more date, last is the one with the least data). To do so, I have to use the value I currently use in the `keys` property, but doing this, the label in the left is changed, and I don’t want this.

![4045B5EC-80FE-493F-AA2D-DAF1BF8903D5](https://user-images.githubusercontent.com/38476041/129019076-ab62af57-88ef-4647-a746-bafb013b21a7.jpeg)

Is there any way that I can use the `carrierId` in the left labels, but still sort my bar chart by the `key` property?

**Comment by wyze** [comment, marked as answer]
We currently don't do any sorting of the data, so if you sort by your value key and pass it to the chart. You should get what you would expect. If you can't get that working, please provide a CodeSandbox and I can help you out further.

**Reply by melarenina** [reply]
It worked, thanks :)

**Reply by mina-ltse** [reply]
Did something change in the meantime? It seems to me it's sorting alphabetically and I cannot change the order.
